plugins/processors/parser/parser.go

```diff
@@ -120,6 +133,20 @@ func merge(base telegraf.Metric, metrics []telegraf.Metric) telegraf.Metric {
 	return base
 }
 
+func mergeWithTimestamp(base telegraf.Metric, metrics []telegraf.Metric) telegraf.Metric {
+	for _, metric := range metrics {
+		for _, field := range metric.FieldList() {
+			base.AddField(field.Key, field.Value)
+		}
+		for _, tag := range metric.TagList() {
+			base.AddTag(tag.Key, tag.Value)
+		}
+		base.SetName(metric.Name())
+		base.SetTime(metric.Time())
```

Comment:
Shouldn't we catch the case where none of the given metrics contain a valid time and retain the original time? If not, we should at least document the behavior...

Reply:
This is a great question. I started to work on a test case for this, each of the parsers I looked at call `time.Now()` and uses that if no timestamp is found. Meaning `metric.Time()` will always be non-zero.

Reply:
Can we still wrap this with a
```golang
		if !m.Time().IsZero() {
			base.SetTime(metric.Time())
		}
```
just in case...

Reply:
sure :) done